dotnet/src/InternalUtilities/src/Diagnostics/ModelDiagnostics.cs

```diff
@@ -136,9 +194,43 @@ private static string ToOpenAIFormat(IEnumerable<ChatMessageContent> chatHistory
 
             sb.Append("{\"role\": \"");
             sb.Append(message.Role);
-            sb.Append("\", \"content\": \"");
+            sb.Append("\", \"content\": ");
             sb.Append(JsonSerializer.Serialize(message.Content));
-            sb.Append("\"}");
+            sb.Append(", \"tool_calls\": ");
+            sb.Append(ToOpenAIFormat(message.Items));
+            sb.Append('}');
+
+            isFirst = false;
+        }
+        sb.Append(']');
+
+        return sb.ToString();
+    }
+
+    /// <summary>
+    /// Convert tool calls to a string aligned with the OpenAI format
+    /// </summary>
+    private static string ToOpenAIFormat(ChatMessageContentItemCollection chatMessageContentItems)
```

Comment:
Although Gemini currently supports function calling, it does not attach the tool call information in the item list.

Reply:
You mean it doesn't add the function calling information to the chat history? If so, that sounds like a bug.
cc: @Krzysztof318

Reply:
Yeah, it doesn't use the `FunctionCallContent` to store function call information in the `ChatMessageContent`. It's not a bug. You can cast the response to a more concrete type (`GeminiChatMessageContent`) to get the function calling stuff. 

I talked to Roger, he said a contributor was planning to setup `FunctionCallContent` in Gemini but the contributor is on vacation. I can add it. Just want to get the OpenAI part done before working on that.